src/torcontrol.cpp

```diff
@@ -271,17 +271,19 @@ static std::map<std::string,std::string> ParseTorReplyMapping(const std::string
     size_t ptr=0;
     while (ptr < s.size()) {
         std::string key, value;
-        while (ptr < s.size() && s[ptr] != '=') {
+        while (ptr < s.size() && s[ptr] != '=' && s[ptr] != ' ') {
             key.push_back(s[ptr]);
             ++ptr;
         }
         if (ptr == s.size()) // unexpected end of line
             return std::map<std::string,std::string>();
+        if (s[ptr] == ' ') // The remaining string is an OptArguments
+            break;
         ++ptr; // skip '='
         if (ptr < s.size() && s[ptr] == '"') { // Quoted string
             ++ptr; // skip opening '"'
             bool escape_next = false;
-            while (ptr < s.size() && (!escape_next && s[ptr] != '"')) {
+            while (ptr < s.size() && (escape_next || s[ptr] != '"')) {
                 escape_next = (s[ptr] == '\\');
                 value.push_back(s[ptr]);
```

Comment:
It should be:
```
if (!escape_next) {
    value.push_back(s[ptr]);
}
```
in order to implement simple unescaping.

This doesn't implement C-style unescaping, but sections 3.21 and 3.24 use QuotedString, not CString, and I am inclined to take that at face value for simplicity, despite what it says in section 2.1.1 (which is only a MAY).